sphinx/testing/_internal/util.py

```diff
@@ -0,0 +1,106 @@
+"""Private utililty functions for :mod:`sphinx.testing.plugin`.
+
+This module is an implementation detail and any provided function
+or class can be altered, removed or moved without prior notice.
+"""
+
+from __future__ import annotations
+
+__all__ = ()
+
+import binascii
+import json
+import os
+import pickle
+from functools import lru_cache
+from typing import TYPE_CHECKING, overload
+
+import pytest
+
+if TYPE_CHECKING:
+    from typing import Any, Final
+
+    from _pytest.nodes import Node as PytestNode
+
+    from sphinx.testing._internal.pytest_util import TestNodeLocation
+
+UID_BITLEN: int = 32
+r"""The bit-length of unique identifiers generated by this module.
+
+Must be a power of two in :math:`[8, 128]`.
+"""
+UID_BUFLEN: Final[int] = UID_BITLEN // 8
+UID_HEXLEN: Final[int] = UID_BITLEN // 4
+
+
+# fmt: off
+@overload
+def make_unique_id() -> str: ...  # NoQA: E704
+@overload
+def make_unique_id(prefix: str | os.PathLike[str]) -> str: ...  # NoQA: E704
+# fmt: on
+def make_unique_id(prefix: str | os.PathLike[str] | None = None) -> str:  # NoQA: E302
+    r"""Generate a random unique identifier prefixed by *prefix*.
+
+    :param prefix: An optional prefix to prepend to the unique identifier.
+    :return: A random unique identifier.
+    """
+    suffix = os.urandom(UID_BUFLEN).hex()
+    return '-'.join((os.fsdecode(prefix), suffix)) if prefix else suffix
+
+
+def get_environ_checksum(*args: Any) -> int:
```

Comment:
Could/should this accept a single (type-hinted) `env` object as a parameter instead?

Reply:
Mmh. Actually the function is very generic, but I can make two versions, where one is typed and expects some explicit parameters and another just takes the kwargs. I don't know if we'll need a more generic thing in the future, that's why I made it like that.